install-app.sh

```
@@ -1,23 +1,156 @@
 #!/bin/bash
 
-# Script to silently install and start the todo web app on the virtual machine. 
-# Note that all commands bellow are without sudo - that's because extention mechanism 
-# runs scripts under root user. 
-
-# install system updates and isntall python3-pip package using apt. '-yq' flags are 
-# used to suppress any interactive prompts - we won't be able to confirm operation 
-# when running the script as VM extention.  
-apt-get update -yq
-apt-get install python3-pip -yq
-
-# Create a directory for the app and download the files. 
-mkdir /app 
-# make sure to uncomment the line bellow and update the link with your GitHub username
-# git clone https://github.com/<your-gh-username>/azure_task_12_deploy_app_with_vm_extention.git
-cp -r devops_todolist_terraform_task/app/* /app
-
-# create a service for the app via systemctl and start the app
-mv /app/todoapp.service /etc/systemd/system/
+# Update system
+export DEBIAN_FRONTEND=noninteractive
```

Add robust shell options to fail fast and avoid subtle errors. Insert near the top (immediately after the shebang) `set -euo pipefail` and optionally `IFS=$'\n\t'`. This prevents the script from continuing after a failed command and makes failures visible to Terraform/VM extension. Right now the script exports DEBIAN_FRONTEND directly without enabling strict mode.